gef.py

```diff
@@ -5937,6 +5946,11 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
             return
 
         # try to establish the remote session, throw on error
+        # Set `.remote` to False here - `GefRemoteSessionManager` invokes code which
+        # calls `is_remote_debug` which checks if `.remote` is None
+        # Change it here early to prevent some spurious errors being thrown while the session
+        # starts up
+        gef.session.remote = False
```

Comment:
Great catch on this! Totally miss this bug!
However, I'm not super fan of this approach (setting it to False) as it introduces a type error. There could be different approaches, not very costly: probably the easiest would to add a state to `GefRemoteSessionManager` (or even just a boolean like `initialized`) that we could check from the outside.

Reply:
Actually considering the remote nature of the manager, yeah a small state attribute could definitely be useful but for this PR, maybe just `initialized` set or something could work.